### PR TITLE
Feature/bitflag checking style 11668.  Fixes #11668

### LINKED ISSUE #11668: Bitflag checking style.
There is some places where bitflags checked like:
`Mask&flag == flag`, but preferred style is `Mask&flag != 0`.
Find all such places and fix them

### PR DESCRIPTION
**Pull request by jimmypuckett**
Replaced the occurrences of "Mask&flag == flag" with "Mask&flag != 0"

Fixes #11668

### CODE REVIEW

**Comment by duglin** [comment]
@jimmypuckett thanks for this.
A couple of things....   first, please squash your commits down to one.  Second, I'm not sure we want to do this.  I may be overly paranoid so I'd want other maintainers to speak up here but this switch only works when the flag is a single bit.  Now, in these cases you found that might be true but it assumes that someone has gone through the trouble of tracking down the actual value of each constant and verified that it is indeed just one bit.  

Now, what worries me is that I tend to think of constants as things that we shouldn't need to examine too closely - meaning we shouldn't need to know their real value. So, let's say some day someone redefines one of those constants to actually be 2 bits instead of one.  This not only breaks this abstraction that I think constants provide but the code as well. I'd prefer if we kept the code not knowing the exact value of the constant and treated it just as a bitmask that could in fact be multiple bits.  But I welcome other's thoughts.

**Comment by LK4D4** [comment]
@duglin I'm sure I want this. It is how it's done in standard library.

**Comment by duglin** [comment]
@LK4D4 as long as we all know that it only works for single-bit masks - which means we need to check each mask now and make sure they never change in the future. Just felt risky to me to have to analyze the exact value of each constant. But if I'm alone in this concern, I'll let it go.

File: daemon/daemon.go [hunk 1]
```diff
@@ -445,8 +445,8 @@ func (daemon *Daemon) setupResolvconfWatcher() error {
 			select {
 			case event := <-watcher.Events:
 				if event.Name == "/etc/resolv.conf" &&
-					(event.Op&fsnotify.Write == fsnotify.Write ||
-						event.Op&fsnotify.Create == fsnotify.Create) {
+					(event.Op&fsnotify.Write != 0 ||
+						event.Op&fsnotify.Create != 0) {
```
Comment (tiborvass) [inline daemon/daemon.go]: these two can be put into one:
`event.Op & fsnotify.Write & fsnotify.Create != 0`

@LK4D4 please confirm

Comment (LK4D4) [inline daemon/daemon.go]: Should work.

Comment (tiborvass) [inline daemon/daemon.go]: @jimmypuckett if you wouldn't mind updating that'd be fantastic!

Comment (jimmypuckett) [inline daemon/daemon.go]: Ill do it tonight. I'm about to eat dinner with the family.

**Comment by LK4D4** [comment]
@jimmypuckett Not a problem. You can do it any time. Also don't forget to eliminate duplicate commits.

File: daemon/daemon.go (hunk 1 above)
Comment (jimmypuckett) [inline daemon/daemon.go]: @tiborvass I am getting the following error when I try to do ```make binary```

```daemon/daemon.go:448: invalid operation: event.Name == "/etc/resolv.conf" && event.Op & fsnotify.Write (mismatched types bool and fsnotify.Op)```

I have the following edit that you requested...

```go
if event.Name == "/etc/resolv.conf" &&
	(event.Op & (fsnotify.Write | fsnotify.Create)) {
```

Comment (tiborvass) [inline daemon/daemon.go]: My bad, I forgot the `!= 0` part !

**Comment by jimmypuckett** [comment]
@LK4D4 I am assuming by duplicate commits that you are wanting one commit with all of the files that I changed?

**Comment by LK4D4** [comment]
@jimmypuckett Yeah, or at least different commit messages. Also would you mind to add `Fixes #11668` to your top-comment? Then github will be automatically close issue on merge. Also next time it is better to write this in body of commit message, which separated from title with one empty line(now body is only your sign).
Here is link not related to your changes, but very interesting and useful: http://chris.beams.io/posts/git-commit/

**Comment by tiborvass** [comment]
@jimmypuckett would you mind running `gofmt -w -s daemon/daemon.go` amend your commit and repush? Thanks!

**Comment by tiborvass** [comment]
LGTM

Ping @LK4D4

**Comment by tiborvass** [comment]
@jimmypuckett I edited your description comment, but for next time make sure you add the "Fixes #xxx" part not in the first line of your commit message (that becomes the title of the PR) but in the lines below (that becomes the description of the PR).

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**